File: apps/backend/app/schemas/ai_review.py
from pydantic import BaseModel


class CodeReviewRequest(BaseModel):
    code: str
    problem_description: str
    language: str = "python"


class CodeReviewResponse(BaseModel):
    task_id: str
    status: str = "pending"
    message: str = "コードレビューを実行中です"


class CodeReviewResult(BaseModel):
    task_id: str
    status: str
    review: str | None = None
    error_message: str | None = None

File: packages/shared-types/src/ai-review.ts
import type { AsyncTaskStatus } from "./common";

export type CodeReviewRequest = {
  code: string;
  problem_description: string;
  language?: string;
};

export type CodeReviewResponse = {
  task_id: string;
  status: AsyncTaskStatus;
  message: string;
};

export type CodeReviewResult = {
  task_id: string;
  status: AsyncTaskStatus;
  review: string | null;
  error_message: string | null;
};
